services/artists-api/src/routes/updateProfileStatusByToken.js

```diff
@@ -0,0 +1,115 @@
+const { StatusCodes, ReasonPhrases } = require('http-status-codes');
+const logger = require('@artistdirectory/logger');
+const generateToken = require('./../utilities/generateToken');
+const emailClient = require('@artistdirectory/email-client');
+const mongodbClient = require('../models/mongodbClient');
+const models = require('../models');
+
+const { DIRECTORY_API_URL } = process.env;
+
+const handler = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  await mongodbClient.connect();
+
+  if (event.source === 'serverless-plugin-warmup') {
+    await new Promise((resolve) => setTimeout(resolve, 25));
+    return 'Lambda is warm!';
+  }
+
+  try {
+    const { reviewToken } = event.pathParameters;
+    const Artist = await models.get('Artist');
+    const { approvalStatus, rejectionReason } = JSON.parse(event.body);
+
+    // generate token for url for editing artist profile if rejected
+    const editProfileToken = await generateToken();
+    const artist = await Artist.findOneAndUpdate(
+      { reviewToken },
+      [
+        {
+          $set: {
+            approvalStatus,
+            editProfileToken: {
+              $cond: {
+                if: approvalStatus === 'rejected',
+                then: editProfileToken,
+                else: null
+              }
+            }
+          }
+        }
+      ],
+      {
+        new: true
+      }
+    );
+    if (approvalStatus === 'rejected') {
+      artist.rejectionReasons.push(rejectionReason);
+    }
+
+    if (!artist) {
+      return {
+        statusCode: StatusCodes.NOT_FOUND,
+        body: ReasonPhrases.NOT_FOUND
+      };
+    }
+
+    const { firstName, lastName, email } = artist;
+
+    // send email to the artist about profile review decision
+    await emailClient.enqueue({
```

Comment:
Let's move this `emailClient.enqueue()` call to `artist.approveProfile()` and `artist.rejectProfile()`. So calling either of these methods would send the appropriate email. Each method would send a different email...like you have ternary statement below.

Reply:
done and thank you for the idea